src/main/java/io/kubernetes/KabaneroClient.java

```diff
@@ -140,24 +144,29 @@ public static String getCLI(ApiClient client, String namespace) throws ApiExcept
     public static void discoverTools(KabaneroToolManager tools) throws IOException, ApiException, GeneralSecurityException {
         ApiClient client = KabaneroClient.getApiClient();
 
+        InputStream inputStream = KabaneroClient.class.getClassLoader().getResourceAsStream("tools.json");
```

Comment:
You might need to close this InputStream at some point. You should look into whether you need to or not, I'm guessing you will need to.